infra/blueprint-test/modules/tft/terraform.go

```diff
@@ -0,0 +1,248 @@
+	logger      *logger.Logger         // custom logger
+	t           testing.TB             // TestingT or TestingB
+}
+
+type option func(*TFBlueprintTest)
+
+func WithName(name string) option {
+	return func(f *TFBlueprintTest) {
+		f.name = name
+	}
+}
+
+func WithFixtureName(fixtureName string) option {
+	return func(f *TFBlueprintTest) {
+		f.fixtureName = fixtureName
+	}
+}
+
+func WithTFDir(tfDir string) option {
+	return func(f *TFBlueprintTest) {
+		f.tfDir = tfDir
+	}
+}
+
+func WithEnvVars(envVars map[string]string) option {
+	return func(f *TFBlueprintTest) {
+		f.envVars = envVars
+	}
+}
+
+func WithSetupPath(setupPath string) option {
+	return func(f *TFBlueprintTest) {
+		f.setupPath = setupPath
+	}
+}
+
+func WithVars(vars map[string]interface{}) option {
+	return func(f *TFBlueprintTest) {
+		f.vars = vars
+	}
+}
+
+func WithLogger(logger *logger.Logger) option {
+	return func(f *TFBlueprintTest) {
+		f.logger = logger
+	}
+}
+
+// Init sets defaults, validates and returns a TFBlueprintTest.
+func Init(t testing.TB, opts ...option) *TFBlueprintTest {
+	tft := &TFBlueprintTest{
+		name: fmt.Sprintf("%s TF Blueprint", t.Name()),
+		t:    t,
+	}
+	// apply options
+	for _, opt := range opts {
+		opt(tft)
+	}
+	// if no custom logger, set default based on test verbosity
+	if tft.logger == nil {
+		tft.logger = utils.GetLoggerFromT()
+	}
+	// one of fixture name or tfDir should be provided
+	if tft.fixtureName != "" && tft.tfDir != "" {
+		t.Fatalf("Only one of FixtureName or TFDir must be provided, found FixtureName=%s, TFDir=%s", tft.fixtureName, tft.tfDir)
+	}
+	// if both fixture name and tfDir are empty, auto discover tfDir based on cwd
+	if tft.fixtureName == "" && tft.tfDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("unable to get wd :%v", err)
+		}
+		tfdir, err := discovery.ConfigDirFromCWD(cwd)
+		if err != nil {
+			t.Fatalf("unable to detect TFDir :%v", err)
+		}
+		tft.tfDir = tfdir
+	}
+	// if fixture name compute tfDir from given fixture name
+	if tft.fixtureName != "" {
+		tfModFixtureDir := getTFModuleFixtureDir(tft.fixtureName)
+		if _, err := os.Stat(tfModFixtureDir); os.IsNotExist(err) {
+			t.Fatalf("TFDir path derived from %s as %s does not exist", tft.fixtureName, tfModFixtureDir)
+		}
+		tft.tfDir = tfModFixtureDir
+	}
+	// setupPath is empty, try known setupPath
+	if tft.setupPath == "" {
+		tfModSetupDir := getTFModuleSetupDir()
+		if _, err := os.Stat(tfModSetupDir); os.IsNotExist(err) {
+			t.Logf("Setup dir %s not found, skipping loading setup outputs as fixture inputs", tfModSetupDir)
+		}
+		tft.setupPath = tfModSetupDir
+	}
+	// load TFEnvVars
+	tfEnvVars := make(map[string]string)
+	//load from setup outputs
+	if tft.setupPath != "" {
+		tft.logger.Logf(tft.t, "Loading env vars from setup %s", tft.setupPath)
+		loadTFEnvVar(tfEnvVars, tft.getTFSetupOPMap())
+	}
+	// load additional env variables
+	if tft.envVars != nil {
+		loadTFEnvVar(tfEnvVars, tft.envVars)
```

Comment:
Having `tft.envVars` start as the unprefixed variables then transform into the prefixed variables is potentially confusing. Either:

1. Use a separate field for the transformed variables
2. Preferably, directly inject the prefix during `WithEnvVars`. Then this whole section can be simplified without needing an intermediate `tfEnvVars` variable.

Reply:
Good idea, I went with injecting it during WithEnvVars.